ReviewChecklist.md

```diff
@@ -0,0 +1,167 @@
+- [ ] **Validate inputs and handle errors**. Note that assertions are **not** validation.
+
+### Readability
+- [ ] **Choose good names.**
+    - [ ] Is the name straightforward to understand?
+        Do you feel the need to jump back to the definition and remind yourself what it was whenever you see it?
+    - [ ] Is the name unambiguous in the context where it is used?
+    - [ ] Avoid abbreviations.
+- [ ] **Source files, classes and public functions should have docstrings.**
+- [ ] **Avoid code duplication.** But not fanatically. Minimal amounts of duplication are acceptable if it aids readability.
+- [ ] **Do not leave dead or commented-out code behind.** You can still see old code in history.
+      If you really have a good reason to do it, always leave a comment explaining what it is and why it is there.
+- [ ] **Mark hacks as such.** If you have to leave behind a temporary workaround, make
+    sure to include a comment that explains why and in what circumstances it can be removed.
+    Preferably link to an issue you reported upstream.
+- [ ] **Avoid obvious comments.**
+- [ ] **Do include comments when the reader may need extra context to understand the code.**
+
+### Commits and PRs
+- [ ] **Avoid hiding functional changes inside refactors.**
+    E.g. when fixing a small bug, or changing user-visible behavior, put the change in a separate commit.
+    Do not mix it with another change that renames things or reformats the code around, making the fix itself hard to identify.
+- [ ] **Whenever possible, split off refactors or unrelated changes into separate PRs.**
+    Smaller PRs are easier and quicker to review.
+    Splitting off refactors helps focus on the main point of the PR.
+
+### Common Pitfalls
+The following points are all covered by the coding style but come up so often that it is worth singling them out here:
+- [ ] **Always initialize value types in the definition,** even if you are sure you will assign them later.
+- [ ] **Use "east const" style.** I.e. `T const*`, not `const T *`.
+- [ ] **Keep indentation consistent.** See our [`.editorconfig`](.editorconfig).
+    - [ ] Tabs for C++. But use them for indentation only. Any whitespace later on the line must consist of spaces.
+    - [ ] 4 spaces for most other file types.
+- [ ] **Use `auto` sparingly.** Only use it when the actual type is very long and complicated or when it is
+    already used elsewhere in the same expression.
+- [ ] **Indent braces and parentheses in a way that makes nesting clear.**
+- [ ] **Use `using namespace` only in `.cpp` files.** Use it for `std` and our own modules.
+    Avoid unnecessary `std::` prefix in `.cpp` files (except for `std::move`).
+- [ ] **Use range-based loops and destructuring.**
+- [ ] **Include any headers you use directly,** even if they are implicitly included through other headers.
+
+## Documentation
+- [ ] **Does the PR update relevant documentation?**
```

Comment:
I have the feeling that something is missing here. What is expected to be exactly done if it is a documentation PR?

Reply:
Maybe it's better to just write `Update relevant documentation` here. I got somehow a bit confused reading that first.

Reply:
I think the part that confused me here was that this item is written as a question.

Reply:
Some of these as written as questions to aid the reviewer. Saying "add this or that" to the reviewer makes less sense.